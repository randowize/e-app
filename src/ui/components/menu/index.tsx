import Component from './menu';

export default Component;
