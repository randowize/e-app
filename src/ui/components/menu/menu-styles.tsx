import styled from 'styled-components';

export const Ul = styled.ul`
  list-style: ${(props: any) => (props.show ? 'block' : 'none')};
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 48px;
  padding: 10px 5px;
  align-items: center;
  background-color: rgba(0, 0, 5, 0.85);
  box-shadow: 5px 0 10px 3px rgba(3, 1, 36, 0.37);
  z-index: 10;
`;

export const Li = styled.li`
  margin: 10px auto;
  font-size: 24px;
  color: #00ffff;
  transition: all 0.5s ease;
  &:hover {
    cursor: pointer;
    color: #00ffff;
    .fa {
      text-shadow: 0 0 10px #2fffff;
    }
  }
  &.active {
    color: #ff0;
  }
`;

export const VALUE = 5;
