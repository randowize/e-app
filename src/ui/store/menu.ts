import { observable, computed, action } from 'mobx';


export default class MenuStore {
    //body
     @observable  icons = [
       {name: 'home', id: 1},
       {name: 'search', id: 2},
       {name: 'rocket', id: 3}
       ];
    @observable menuClass = 'has-options';
    @observable activeMenuItem: number = 1;

    @computed get menuItems(){
        return this.icons;
    }

    @computed get activeItem(){
      return this.activeMenuItem;
    }

    @action onMenuItemClick = (menuInfo: any) => {
        if (this.menuClass === 'has-options') {
            if (this.activeMenuItem === menuInfo.id) {
                this.setMenuClass('');
                return;
            }
        }else {
             this.setMenuClass('has-options');
        }
        this.activeMenuItem = menuInfo.id;
    }

    @action setMenuClass = (mc) => {
        this.menuClass = mc;
    }

    @computed get menuType(){
        return this.menuClass;
    }

};
