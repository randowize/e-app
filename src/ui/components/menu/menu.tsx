import * as React from 'react';
import * as Fa from 'react-fontawesome';
import {selectProps} from '../../../utils/props-selector';
import {Ul, Li} from './menu-styles';

export interface IProps {
    [key: string] : any;
    menuItems: any[];
}

const Menu: React.StatelessComponent < IProps > = (props) => (
    <Ul>
        {props
            .menuItems
            .map(item => (
                <Li
                    className={props.activeItem === item.id
                    ? 'active'
                    : ''}
                    onClick={() => props.onMenuItemClick(item)}
                    key={item.id}>
                    <Fa name={item.name}/>
                </Li>
            ))
}
    </Ul>
);

Menu.defaultProps = {
    menuItems: []
};

const props = ['menuItems', 'onMenuItemClick', 'activeItem'];

export default selectProps('menuStore')(...props)(Menu);
