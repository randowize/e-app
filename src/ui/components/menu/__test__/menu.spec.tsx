import * as React from 'react';
import * as rtr  from 'react-test-renderer';
import {Menu} from '../menu';
it('renders correctly', () => {
  const output = rtr.create(<Menu menuItems={[{name: 'bus', id: '1'}]}/>).toJSON();
  expect(output).toMatchSnapshot();
});