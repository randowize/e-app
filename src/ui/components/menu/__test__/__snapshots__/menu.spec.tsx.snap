// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="sc-bdVaJa iVqtFN"
>
  <li
    className="sc-bwzfXH dILcNu"
  >
    <img
      src="E:\\\\fsd\\\\electron-app\\\\src\\\\ui\\\\components\\\\menu\\\\logo-1.png"
      style={
        Object {
          "borderRadius": "50%",
        }
      }
    />
  </li>
  <li
    className="sc-bwzfXH dILcNu"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-bus"
    />
  </li>
</ul>
`;
